scala-effect-api/src/main/scala/com/avast/metrics/scalaeffectapi/Timer.scala

```diff
@@ -0,0 +1,27 @@
+package com.avast.metrics.scalaeffectapi
+
+import cats.Monad
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import com.avast.metrics.api.Timer.TimeContext
+
+import java.time.Duration
+
+trait Timer[F[_]] {
+  def start: F[TimeContext]
+  def stop(context: TimeContext): F[Duration]
+}
+
+object Timer {
+  object Dsl {
+    implicit class TimerOps[F[_]](val timer: Timer[F]) extends AnyVal {
+      def time[A](block: F[A])(implicit monad: Monad[F]): F[A] = {
+        for {
+          context <- timer.start
+          result <- block
+          _ <- timer.stop(context)
+        } yield result
+      }
+    }
+  }
+}
```

Comment:
does this need to be in companion object? I'd suggest it doesn't add much being there and makes it more difficult to use - i would just slap it into the interface itself (same for the TimerPair below)

Reply:
After my last change (generic context removel), there is no need for that. I'll move it.